tests/core/input/test_input_event.h

```diff
@@ -0,0 +1,115 @@
+/* Copyright (c) 2014-present Godot Engine contributors (see AUTHORS.md). */
+/* Copyright (c) 2007-2014 Juan Linietsky, Ariel Manzur.                  */
+/*                                                                        */
+/* Permission is hereby granted, free of charge, to any person obtaining  */
+/* a copy of this software and associated documentation files (the        */
+/* "Software"), to deal in the Software without restriction, including    */
+/* without limitation the rights to use, copy, modify, merge, publish,    */
+/* distribute, sublicense, and/or sell copies of the Software, and to     */
+/* permit persons to whom the Software is furnished to do so, subject to  */
+/* the following conditions:                                              */
+/*                                                                        */
+/* The above copyright notice and this permission notice shall be         */
+/* included in all copies or substantial portions of the Software.        */
+/*                                                                        */
+/* THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,        */
+/* EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF     */
+/* MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. */
+/* IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY   */
+/* CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,   */
+/* TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE      */
+/* SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.                 */
+/**************************************************************************/
+
+#ifndef TEST_INPUT_EVENT_H
+#define TEST_INPUT_EVENT_H
+
+#include "core/input/input_event.h"
+#include "core/math/rect2.h"
+#include "core/os/memory.h"
+#include "core/variant/array.h"
+
+#include "tests/test_macros.h"
+
+namespace TestInputEvent {
+TEST_CASE("[InputEvent] Signal is emitted when device is changed") {
+	Ref<InputEventKey> inputEvent;
+	inputEvent.instantiate();
+
+	SIGNAL_WATCH(*inputEvent, SNAME("changed"));
+	Array args1;
+	Array empty_args;
+	empty_args.push_back(args1);
+
+	inputEvent->set_device(1);
+
+	SIGNAL_CHECK("changed", empty_args);
+	CHECK(inputEvent->get_device() == 1);
+
+	SIGNAL_UNWATCH(*inputEvent, SNAME("changed"));
+}
+
+TEST_CASE("[InputEvent] Test accumulate") {
+	Ref<InputEventMouseMotion> iemm1, iemm2;
+	Ref<InputEventKey> iek;
+
+	iemm1.instantiate(), iemm2.instantiate();
+	iek.instantiate();
+
+	iemm1->set_button_mask(MouseButtonMask::LEFT);
+
+	CHECK_FALSE(iemm1->accumulate(iemm2));
+
+	iemm2->set_button_mask(MouseButtonMask::LEFT);
+
+	CHECK(iemm1->accumulate(iemm2));
+
+	CHECK_FALSE(iemm1->accumulate(iek));
+	CHECK_FALSE(iemm2->accumulate(iek));
+}
+
+TEST_CASE("[InputEvent][SceneTree] Test methods that interact with the InputMap") {
+	const String MOCK_ACTION = "mock_action";
```

Comment:
Please use snake_case for varibale names in the c++ source code and rename `MOCK_ACTION`.